Create Shehzaib.py

msali3 opened the pull request:
...

msali3 committed:
Create Shehzaib.py

...

msali3 committed:
changes

Please accept these changes

msali3 reviewed (commented):
ok

chaedri commented on activities/FireSpotting.json:
```diff
@@ -0,0 +1,20 @@
+      "layers": [
+        ["d.rast", "map=fire_hotspots"]
+      ],
+      "base": "base_elevation_layer",
```
```suggestion
      "base": "dem",
```

chaedri commented on activities/FireSpotting.py:
```diff
@@ -0,0 +1,52 @@
+                        map='fire_hotspots',
+                        env=env)
+
+def main():
```
This function only runs during testing so, if you're developing in the Tangible Landscape Tutorial Notebook, you can use the example dataset (i.e. "elev_lid972_1m")

chaedri commented on activities/FireSpotting.py:
```diff
@@ -0,0 +1,52 @@
+    # Updated layer names
+    base_elev = 'base_elevation_layer'
+    scanned_elev = 'scanned_elevation'
+    spotting_threshold = 1  # The threshold for fire spotting depends on height can be adjusted for canopy
```
This threshold should probably be hardcoded into the function so since TL won't run this section of code during the activity.

chaedri commented on activities/FireSpotting.py:
```diff
@@ -0,0 +1,52 @@
+                          stdin="-100 black\n-20 red\n0 white\n20 blue\n100 black",
+                          env=env)
+
+def simulate_fire_spotting(difference_layer, spotting_threshold, env):
```
```suggestion
def run_simulate_fire_spotting(difference_layer, spotting_threshold, env):
```

chaedri commented on activities/FireSpotting.py:
```diff
@@ -0,0 +1,52 @@
+        threshold=spotting_threshold),
+        env=env)
+    
+    gscript.run_command('d.rast',
```
This is taken care of in the JSON file. No need to call `d.rast` here.

chaedri commented:
Try installing black to get the code quality checks to pass:

https://black.readthedocs.io/en/stable/

msali3 committed:
Tangible Landscape by Shehzaib

Hi, I have made changes to my project on Tangible Landscape. Please let me know if it works?

msali3 committed:
Merge branch 'master' of https://github.com/msali3/gis714-2024-tangible-landscape

petrasovaa commented:
> Try installing black to get the code quality checks to pass:
> 
> https://black.readthedocs.io/en/stable/

Please do that.

petrasovaa commented:
> > Try installing black to get the code quality checks to pass:
> > https://black.readthedocs.io/en/stable/
> 
> Please do that or use https://black.vercel.app/ online tool.

petrasovaa commented on activities/FireSpotting.json:
```diff
@@ -0,0 +1,28 @@
+          "interpolate": true
+        },
+        "calibrate": true,
+        "analyses": "FireSpotting3.py",
```
```suggestion
        "analyses": "FireSpotting.py",
```

msali3 committed:
Update FireSpotting.py

msali3 committed:
Merge branch 'master' of github.com:msali3/gis714-2024-tangible-landscape

msali3 committed:
Update FireSpotting.json

msali3 committed:
Update FireSpotting.py

msali3 committed:
Update FireSpotting.py

petrasovaa commented:
This looks differently than what I saw in the class, could you explain what are you trying to do?

petrasovaa commented:
> This looks differently than what I saw in the class, could you explain what are you trying to do?

There are several problems with this, the difference you compute won't work easily in the github continuous integration, so I am trying to better understand your goal first  to be able to help.

petrasovaa commented:
In case you would want to use the marker detection, start with the advanced_example.py.

msali3 committed:
changes

msali3 committed:
changes added

msali3 commented:
> > This looks differently than what I saw in the class, could you explain what are you trying to do?
> 
> There are several problems with this, the difference you compute won't work easily in the github continuous integration, so I am trying to better understand your goal first to be able to help.

msali3 committed:
Update FireSpotting.json

msali3 commented:
> This looks differently than what I saw in the class, could you explain what are you trying to do?

It is similar to what i was trying to do before but it became rather complicated, i will try reusing the code we worked on in class.

msali3 committed:
fixing CRFL to LF

msali3 committed:
Update FireSpotting.json

msali3 committed:
trying to fix the newline error

Updating the .json file

msali3 committed:
Merge branch 'master' into master
